### PR TITLE
Only active cells

### PR DESCRIPTION
**Pull request by joakim-hove**
Started to *actually use* the new 3D proeprty implementation. This is for now - mainly a placeholder.

Downstream:
https://github.com/OPM/opm-material/pull/380
https://github.com/OPM/opm-simulators/pull/2213

### CODE REVIEW

**Comment by alfbr** [comment]
I am unable to build this

**Comment by joakim-hove** [comment]
Have you turned off testing?

**Comment by alfbr** [comment]
Yes, all should be according to the recipe. Please test at your end.

**Comment by joakim-hove** [comment]
Ok - I'll test.

**Comment by joakim-hove** [comment]
OK;
```
bash% cmake .. -DENABLE_3DPROPS_TESTING=ON -DBUILD_TESTING=OFF ; make -j 16
```
works on my machine.

**Comment by blattms** [comment]
Maybe this needs some PRs of other modules?

**Comment by joakim-hove** [comment]
> Maybe this needs some PRs of other modules?

yes - absolutely; it is https://github.com/OPM/opm-simulators/pulls which is the starting point

**Comment by alfbr** [comment]
My bad, a syntax mistake in a cmake switch, thanks for your patience. Will report back once we get it tested.

File: src/opm/output/eclipse/WriteInit.cpp [hunk 1]
```diff
@@ -423,79 +423,41 @@ namespace {
...
-
-            write(prop, grid.compressedVector(dflt),
-                  opm_property.compressedCopy(grid));
-#endif
         }
     }
 
     template <typename T, class WriteVector>
     void writeCellPropertiesValuesOnly(const Properties& propList,
-                                       const ::Opm::GridProperties<T>&
```
Comment (bska) [inline src/opm/output/eclipse/WriteInit.cpp]: This function is now only used for floating-point properties so you can remove the template argument `T` and use `has<double>()` and `get<double>()` in the function body.

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=2213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=2213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=2213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=2213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 update_data please

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 update_data please

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 update_data please

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 please

**Comment by joakim-hove** [comment]
I don't really understand  current [jenkins failure](https://ci.opm-project.org/job/opm-common-PR-builder/1201/) - someone take a peek?

**Comment by atgeirr** [comment]
> I don't really understand current [jenkins failure](https://ci.opm-project.org/job/opm-common-PR-builder/1201/) - someone take a peek?

The failure is due to an upstream change in opm-material requiring one more template argument for BlackoilFluidState (to enable/disable brine). The corresponding downstream change in opm-simulators is of course not part of OPM/opm-simulators#2213. I would think that that PR would be merged with the current master for testing, making it unnecessary to rebase it just for this, but perhaps I am wrong. Maybe that only applies to the current (i.e. opm-common) repo? @akva2 do you have any info on this?

In the meantime, rebasing the two downstream PRs should be sufficient. Also, is there not an update-data PR we also must include? I cannot find that now?!

**Comment by joakim-hove** [comment]
> In the meantime, rebasing the two downstream PRs should be sufficient.

Hmm - I thought everything was rebased; done now at least.

> Also, is there not an update-data PR we also must include? 

Yes it is - I just wanted to rerun everything and verify that it is still only the same test failing.

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 please

**Comment by joakim-hove** [comment]
> My earlier comment about writeCellPropertiesWithDefaultFlag()

Fixed now - at last.

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 please

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 update_data please

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 opm-tests=229 please

**Comment by joakim-hove** [comment]
jenkins build this opm-material=380 opm-simulators=2213 opm-tests=229 please

**Pull request merged**